### DISCUSSION TITLE
Can you autogenerate fragments for you base model?

### DISCUSSION THREAD

**Discussion by brentshulman-silkline** [dotansimha/graphql-code-generator#9914] [General] [answered]
Ex: if I have a GQL model with a few fields, can I get it to autogenerate a fragment with all the fields? It would make querying, in some cases, easier.

**Comment by PaulRBerg** [comment]
Auto-generating fragments would be super helpful!

Can one of the contributors please comment here?

@dotansimha @ardatan @n1ru4l @eddeee888

**Comment by eddeee888** [comment]
Hello 👋 
Could you help me understand the use case and maybe some examples of how the workflow would work ?

**Reply by PaulRBerg** [reply]
Hi! Thanks for the quick reply.

The use case is to simply not have to maintain code like this:

```graphql
fragment ContractFragment on Contract {
  id
  address
  version
}
```

When the `Contract` type is defined like so in my GraphQL schema:

```graphql
type Contract @entity {
  id: String!
  address: Bytes!
  version: String!
}
```

I want to auto-generate the `ContractFragment` fragment without manually maintaining the `fragment` definition.

**Reply by eddeee888** [reply]
Hi @PaulRBerg , 
This seems like it'd automatically fetch new fields when codegen is run on the client?

What happens if there's a nested relationship? e.g.

```graphql
type Contract @entity {
  id: String!
  address: Bytes!
  version: String!
  relatedContract: Contract!
}
```

**Reply by PaulRBerg** [reply]
Yes, new fields would be automatically fetched.

Maybe those can be excluded by default and the user can be informed about nested relationships via a console warning?

**Reply by eddeee888** [reply, marked as answer]
Is there a reason why new fields should be automatically fetched? GraphQL's default mode of operation is "fetch data when you need it". Adding new field/s when they are available in the schema can easily lead to over-fetching (unless I misunderstood the use cases here).

If this feature is implemented into the core plugins, I can see these follow-up requirements (could be complicated to implement, but doesn't help with the default mode of operation):
- Must avoid nested types
- Option to avoid adding some fields to the fragment

Most likely this will not be implemented into the core plugins, but feel free to create your own plugin to suit your use cases 🙂

**Reply by PaulRBerg** [reply]
Yeah fair enough, a community plugin sounds like the right way to go about this.
